### ISSUE TITLE
alternative optimization score

### ISSUE THREAD

**Issue by stereohelix** [gekkowarez/gekkoga#23]
I'm interested in optimizing the sharpe ratio, or weighing sharpe optimization against profit. Seems like this would be a useful feature.

**Comment by talentoscope** [comment]
Check develop branch ;)

**Comment by stereohelix** [comment]
Excellent, I ran a few tests today with good results. The new index.js also provides an easy way to edit how score is calculated. 

I think the next step would be to send gekko a riskFreeReturn value from the gekkoga config.

**Comment by stereohelix** [comment]
Fun surprise this morning: 6 trades with a sharpe ratio of 2384! I had no idea that was even possible, I think that puts the standard deviation at about 0.00014% profit.

Which lead me to a new way of treating score:

         populationScores[i] = populationProfits[i] * (1 + Math.atan(populationSharpes[i]));

Basically this says that I only really care about the details of the Sharpe ratio if its on the scale of 1.0. Anything too far from that makes the impact roll off.

Which lead me to another thought. I would also be handy to take into account exposure time. I'm not sure how to pull that from Gekko, but the score would look something like: 

         populationScores[i] = populationProfits[i] * (1 + Math.atan(populationSharpes[i]) / (1 + populationExposure[i]));

Basically the same idea, except I want to greatest return with the least total exposure, with more impact on the score the closer it is to 1 (or 100%).

**Comment by talentoscope** [comment]
Very clever! Will be cool to see what happens from this. I too have had really high Sharpe ratios, which are not meaningful in any way if every trade is a positive ROI. Still useful for optimising though.

**Comment by carlosatta** [comment]
Hi. I'm interesed about this change: can you share more information about your index.js (ex.: patch)? Tnks

**Comment by stereohelix** [comment]
Hi Carlos, 

I've refined this a bit since writing. Score should be something in the form of 

scores = Math.tanh(Profits / targetProfits) * Math.tanh(Sharpes / targetSharpes) * Math.tanh(targetExposure / Exposure)

This essentially limits the score according to the order of magnitude of any ratio of results to targets.

For example, my current index.js is attached as index.txt. But I think this is also a general approach to avoiding all sorts of over-optimization.

[index.txt](https://github.com/gekkowarez/gekkoga/files/1669522/index.txt)

**Comment by generalectric** [comment]
@stereohelix, that's awesome! How are your results are they improved over the current gekkoga functionality, I'm assuming? If so would you like to create a PR to improve gekkoga?

**Comment by stereohelix** [comment]
Yes, it reaches reasonable targets, usually in a few minutes, and them improves them proportionally.

Just figured out how to do a pull request. Submitted!
